routes/webauthn.go

```diff
@@ -0,0 +1,377 @@
+package routes
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/httprate"
+	conf "github.com/muety/wakapi/config"
+	"github.com/muety/wakapi/middlewares"
+	"github.com/muety/wakapi/models"
+	routeutils "github.com/muety/wakapi/routes/utils"
+	"github.com/muety/wakapi/services"
+)
+
+const (
+	// HTTP constants
+	contentTypeHeader   = "Content-Type"
+	applicationJSONType = "application/json"
+
+	// Error message constants
+	invalidJSONMessage     = "invalid JSON"
+	internalServerErrorMsg = "internal server error"
+)
+
+type WebAuthnHandler struct {
+	config   *conf.Config
+	userSrvc services.IUserService
+}
+
+func NewWebAuthnHandler(userService services.IUserService) *WebAuthnHandler {
+	return &WebAuthnHandler{
+		config:   conf.Get(),
+		userSrvc: userService,
+	}
+}
+
+func (h *WebAuthnHandler) RegisterRoutes(router chi.Router) {
+	if !h.config.Security.WebAuthnEnabled {
+		return
+	}
+
+	webAuthnRouter := chi.NewRouter()
+	webAuthnRouter.Use(httprate.LimitByRealIP(h.config.Security.GetLoginMaxRate()))
+
+	// Registration endpoints (require authentication)
+	webAuthnRouter.Group(func(r chi.Router) {
+		r.Use(middlewares.NewAuthenticateMiddleware(h.userSrvc).Handler)
+		r.Get("/credentials", h.GetCredentials)
+		r.Post("/register/begin", h.PostRegisterBegin)
+		r.Post("/register/finish", h.PostRegisterFinish)
+		r.Delete("/credentials/{credentialId}", h.DeleteCredential)
+	})
+
+	// Authentication endpoints (public)
+	webAuthnRouter.Post("/login/begin", h.PostLoginBegin)
+	webAuthnRouter.Post("/login/finish", h.PostLoginFinish)
+
+	router.Mount("/auth/webauthn", webAuthnRouter)
+}
+
+// PostRegisterBegin starts WebAuthn credential registration for authenticated users
+func (h *WebAuthnHandler) PostRegisterBegin(w http.ResponseWriter, r *http.Request) {
+	user := middlewares.GetPrincipal(r)
+	if user == nil {
+		http.Error(w, "unauthorized", http.StatusUnauthorized)
+		return
+	}
+
+	var req struct {
+		CredentialName string `json:"credentialName,omitempty"`
+	}
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "failed to read request body", http.StatusBadRequest)
+		return
+	}
+
+	if len(body) > 0 {
+		if err := json.Unmarshal(body, &req); err != nil {
+			http.Error(w, invalidJSONMessage, http.StatusBadRequest)
+			return
+		}
+	}
+
+	options, sessionData, err := h.userSrvc.WebAuthnBeginRegistration(user, req.CredentialName)
+	if err != nil {
+		conf.Log().Request(r).Error("WebAuthn registration begin failed", "error", err)
+		http.Error(w, "failed to begin registration", http.StatusInternalServerError)
+		return
+	}
+
+	// Store session data in the response for the frontend to include in the finish request
+	response := map[string]interface{}{
+		"options":     options,
+		"sessionData": sessionData,
```

Comment:
Why is `sessionData` a stirng-serialized JSON object _inside_ a JSON object? This seems odd. I'd prefer a response format similar to the implementation at https://webauthn.io/.